### PR TITLE
check_format: use GOPATH and add install commands for formatting libraries

### PR DESCRIPTION
**Pull request by cmluciano**
*title*: check_format: use GOPATH and add install commands for formatting libraries

*Description*:
The buildifier docs recommend using the go toolchain to install buildifier. I'm not sure if current default just reflects the original authors GOPATH.

I added explicit installation instructions else `./tools/check_format.sh` may fail for a new contributor with errors that seemingly indicate an actual format error.

*Risk Level*: Low 

*Testing*:
passing locally with `./tools/check_format.sh check`

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -53,9 +54,10 @@ Envoy compiles and passes tests with the version of clang installed by XCode 8.3
 Apple LLVM version 8.1.0 (clang-802.0.42).
 
 3. Install Golang on your machine. This is required as part of building [BoringSSL](https://boringssl.googlesource.com/boringssl/+/HEAD/BUILDING.md)
-and also for [Buildifer](https://github.com/bazelbuild/buildtools) which is used for formatting bazel BUILD files.
-4. `bazel fetch //source/...` to fetch and build all external dependencies. This may take some time.
-5. `bazel build //source/exe:envoy-static` from the Envoy source directory.
+and also for [Buildifer](https://github.com/bazelbuild/buildtools) which is used for formatting bazel BUILD files. 
+4. `go get go get github.com/bazelbuild/buildtools/buildifier` to install buildifier
```
Comment (danielhochman) [inline bazel/README.md]: repeated `go get`

**Comment by danielhochman** [comment]
nice. i was lazy and symlinked buildifier into `/usr/lib/go` when i ran into this issue, thanks for fixing.

**Comment by cmluciano** [comment]
@danielhochman :P looks like this was also being leveraged by a container somewhere, looking where now

**Comment by htuch** [comment]
Please merge master to pick up https://github.com/envoyproxy/envoy/pull/2613, this should fix CI TSAN.

**Comment by cmluciano** [comment]
ok fingers crossed on merging of master

File: bazel/README.md [hunk 2]
```diff
@@ -53,9 +54,10 @@ Envoy compiles and passes tests with the version of clang installed by XCode 8.3
 Apple LLVM version 8.1.0 (clang-802.0.42).
 
 3. Install Golang on your machine. This is required as part of building [BoringSSL](https://boringssl.googlesource.com/boringssl/+/HEAD/BUILDING.md)
-and also for [Buildifer](https://github.com/bazelbuild/buildtools) which is used for formatting bazel BUILD files.
-4. `bazel fetch //source/...` to fetch and build all external dependencies. This may take some time.
-5. `bazel build //source/exe:envoy-static` from the Envoy source directory.
+and also for [Buildifer](https://github.com/bazelbuild/buildtools) which is used for formatting bazel BUILD files. 
+4. `go get github.com/bazelbuild/buildtools/buildifier` to install buildifier
```
Comment (htuch) [inline bazel/README.md]: Why only on Apple?

Comment (cmluciano) [reply bazel/README.md]: Should we split the OS stuff out into a unique heading?

Comment (htuch) [reply bazel/README.md]: Never mind, I was reading this incorrectly via the GH diffs, looks good.

**Pull request merged**